### PR TITLE
Inform user when OpenID login fails

### PR DESCRIPTION
**Pull request by mudler**
When OpenID provider returns messages that cannot be recognized by
Net::OpenID::Consumer, inform the user about the failed login instead of
returning 500.

Related Progress issue: https://progress.opensuse.org/issues/20218

### CODE REVIEW

File: lib/OpenQA/WebAPI/Auth/OpenID.pm [hunk 1]
```diff
@@ -105,9 +105,16 @@ sub auth_response {
...
+        my ($err, $txt) = @_;
+        $self->app->log->error($err, $txt);
+        $self->flash(error => "$err: $txt");
+        return (error => 0);
+    };
+
     $csr->handle_server_response(
         not_openid => sub {
-            die "Not an OpenID message";
+            return $err_handler->("Failed to login", "OpenID provider returned invalid data. Please retry again later");
```
Comment (okurz) [inline lib/OpenQA/WebAPI/Auth/OpenID.pm]: Why should we retry later? I think the problem is not a temporary one but a broken openID server. Sure, the problem is most likely not within openQA. Can we give more information to the user?

Comment (mudler) [inline lib/OpenQA/WebAPI/Auth/OpenID.pm]: As you underlined the problem seems to lay in the OpenID provider, but Net::OpenID::Consumer will bail out if we don't cover the error https://metacpan.org/source/WROG/Net-OpenID-Consumer-1.18/lib/Net/OpenID/Consumer.pm#L440 . Looks like we hit `not_openid` when providers returns no message, or can't be parsed by the consumer. @okurz what kinda of message do you suggest here?

Comment (okurz) [inline lib/OpenQA/WebAPI/Auth/OpenID.pm]: Hm, I don't know in detail but something that points the user to the real problem and a suggestion how to continue. I don't think "Please retry again later" will help. But when just waiting worked for you then it's fine to keep it as is for now :-)

**Comment by coolo** [comment]
please rebase

**Comment by coolo** [comment]
taking the risk

**Pull request merged**